financial-connections-example/src/main/java/com/stripe/android/financialconnections/example/FinancialConnectionsPlaygroundViewModel.kt

```diff
@@ -49,42 +53,31 @@ class FinancialConnectionsPlaygroundViewModel(
         }
     }
 
-    fun startFinancialConnectionsSession(
-        merchant: Merchant,
-        flow: Flow,
-        keys: Pair<String, String>,
-        email: String
-    ) {
+    fun startFinancialConnectionsSession() = with(state.value.settings) {
         _state.update {
             it.copy(
                 status = emptyList(),
                 emittedEvents = emptyList()
             )
         }
-        when (flow) {
-            Flow.Data -> startForData(merchant, keys, email.takeIf { it.isNotEmpty() })
-            Flow.Token -> startForToken(merchant, keys, email.takeIf { it.isNotEmpty() })
-            Flow.PaymentIntent -> startWithPaymentIntent(
-                merchant,
-                keys,
-                email.takeIf { it.isNotEmpty() }
-            )
+        Log.d(
+            "FinancialConnections",
+            "Starting session with settings: ${asJsonString()}"
+        )
+        saveToSharedPreferences(getApplication())
+        when (state.value.flow) {
+            Flow.Data -> startForData(this)
+            Flow.Token -> startForToken(this)
+            Flow.PaymentIntent -> startWithPaymentIntent(this)
         }
     }
 
-    private fun startWithPaymentIntent(
-        merchant: Merchant,
-        keys: Pair<String, String>,
-        email: String?
-    ) {
+    private fun startWithPaymentIntent(snapshot: PlaygroundSettings) {
```

Comment:
There's a few of these, but you're calling this `snapshot` which I don't think you're using (see following methods that do the same thing).

Reply:
https://github.com/stripe/stripe-android/pull/7448/commits/c7a2cba1997a2ea4cbb577621883a65d75a3e265